zerver/lib/test_classes.py

```diff
@@ -230,7 +224,7 @@ def run(self, result: Optional[TestResult] = None) -> Optional[TestResult]:  # n
     DEFAULT_SUBDOMAIN = "zulip"
     TOKENIZED_NOREPLY_REGEX = settings.TOKENIZED_NOREPLY_EMAIL_ADDRESS.format(token="[a-z0-9_]{24}")
 
-    def set_http_headers(self, kwargs: Dict[str, ClientArg]) -> None:
```

Comment:
What I have been doing essentially is to make `kwargs` a homogenous dictionary of `str`. So I split out flags like `skip_user_agent` here.